src/messaging/Message.utils.ts

```diff
@@ -0,0 +1,297 @@
+import { CompressedAttestedClaim } from '../types/AttestedClaim'
+import {
+  QuoteUtils,
+  ClaimUtils,
+  AttestedClaimUtils,
+  MessageBody,
+  AttestationUtils,
+  MessageBodyType,
+  IAttestedClaim,
+  RequestForAttestationUtils,
+} from '..'
+import ITerms, { CompressedTerms } from '../types/Terms'
+import * as SDKErrors from '../errorhandling/SDKErrors'
+import {
+  CompressedRejectedTerms,
+  CompressedRequestAttestationForClaim,
+  CompressedSubmitAttestationForClaim,
+  CompressedRequestClaimsForCTypes,
+  CompressedRequestAcceptDelegation,
+  CompressedSubmitAcceptDelegation,
+  CompressedDelegationData,
+  CompressedInformCreateDelegation,
+  IRequestingClaimsForCTypes,
+  IPartialClaim,
+  IRequestingAttestationForClaim,
+  ISubmittingAttestationForClaim,
+  IRequestingAcceptDelegation,
+  ISubmitingAcceptDelegation,
+  IDelegationData,
+  IInformingCreateDelegation,
+  CompressedMessageBody,
+} from './Message'
+
+/**
+ * [STATIC] Compresses a [[Message]] depending on the message body type.
+ *
+ * @param body The body of the [[Message]] which depends on the [[MessageBodyType]] that needs to be compressed.
+ *
+ *
+ */
+
+export function compressMessage(body: MessageBody): CompressedMessageBody {
+  switch (body.type) {
+    case MessageBodyType.REQUEST_TERMS: {
+      const compressedContents = ClaimUtils.compress(body.content)
+      return [body.type, compressedContents]
```

Comment:
But, each of these returns different types. Same with the `decompressMessage` unless I am missing something it won't work since we use the `CompresssedMessageBody` that will have different content and types.

Reply:
the type checking will definitely complain, especially because it doesn't handle switch very well. We could of course circumvent it with `as CompressedMessageBody` after the return statement